mod_reforged/hooks/skills/skill.nut

```diff
@@ -5,4 +5,57 @@
 	{
 		return this.isAttack() && !this.isRanged() && this.getBaseValue("ActionPointCost") <= 4 && this.getBaseValue("MaxRange") == 1;
 	}
+
+	local getHitFactors = o.getHitFactors;
+	o.getHitFactors = function( _targetTile )
+	{
+		local ret = getHitFactors(_targetTile);
+		for (local index = (ret.len() - 1); index >= 0; index--)	// We traverse it in reverse because we want to remove entries during the loop
+		{
+			switch (ret[index].text)
+			{
+				// For these entries we switch out the Icon for a better one
+				case "Immune to stun":
+				case "Immune to being rooted":
+				case "Immune to being disarmed":
+				case "Immune to being knocked back or hooked":
+				ret[index].icon = "ui/tooltips/warning.png";	// These cases should be a warnings and not hide in-between hit-chance lines
+					break;
+
+				// We remove these entries from ever displaying because they are mostly redundant and take away space
+				case "Fast Adaption":		// This should be handled by the perk itself if it really feels like this information is important
+				case "Armed with shield":
+				case "Shieldwall":
+				case "Riposte":		// This has nothing to do with hitchance and there is already a mini-icon for it anyways
+				case "Resistance against ranged weapons":	// This was never followed through for anything other than skeletons
+				case "Resistance against piercing attacks":	// This was never followed through for anything other than skeletons
+				case "Nighttime":
+					ret.remove(index);
+					break;
+
+				// We add additional hit-chance numbers for some other entries
+				case "Too close":
+				case this.getName():
+					if (this.m.HitChanceBonus == 0) break;	// MSU pushes a raw 'this.getName()' string sometimes even when HitChanceBonus is 0. Currently this is with ranged attacks based on AdditionalAccuracy
+					if (this.m.HitChanceBonus < 0)
+					{
+						ret[index].text = ::MSU.Text.colorRed(this.m.HitChanceBonus + "% ") + ret[index].text;
+					}
+					else
+					{
+						ret[index].text = ::MSU.Text.colorGreen(this.m.HitChanceBonus + "% ") + ret[index].text;
+					}
+					break;
+
+				case "Height disadvantage":
+					ret[index].text = ::MSU.Text.colorRed((::Const.Combat.LevelDifferenceToHitMalus * (this.m.Container.getActor().getTile().Level - _targetTile.Level)) + "% ") + ret[index].text;
```

Comment:
You probably need to multiply by -1 to remove the negative sign here (negative sign should be removed because the icon is already showing a negative sign).